File: Algorithms/Algorithms.UnitTests/BubbleSortTests.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Algorithms.Core;
using System.Collections.Generic;

namespace Algorithms.UnitTests
{
    [TestClass]
    public class BubbleSortTests
    {
        [TestMethod]
        public void TestBubbleSort()
        {
            List<int> l = new List<int>();
            l.Add(50);
            l.Add(20);
            l.Add(10);
            l.Add(600);
            l.Add(30);                        

            BubbleSort.Sort(l);
            Assert.AreEqual(10, l[0]);
            Assert.AreEqual(20, l[1]);
            Assert.AreEqual(30, l[2]);
            Assert.AreEqual(50, l[3]);
            Assert.AreEqual(600, l[4]);
        }

        [TestMethod]
        public void TestBubbleSort2()
        {
            List<int> l = new List<int>();
            l.Add(50);
            l.Add(20);
            l.Add(10);
            l.Add(600);
            l.Add(30);

            BubbleSort2.Sort(l);
            Assert.AreEqual(10, l[0]);
            Assert.AreEqual(20, l[1]);
            Assert.AreEqual(30, l[2]);
            Assert.AreEqual(50, l[3]);
            Assert.AreEqual(600, l[4]);
        }
    }
}

File: Python/algorithms/tests/bubble_sort_tests.py
import unittest
from algorithms.core import bubble_sort

class BubbleSortTests(unittest.TestCase):
    """def test_bubble_sort(self) :
        a = [10, 25, 15, 2, 13, 1, 5]
        bubble_sort.bubble_sort(a);
        self.assertEqual([1, 2, 5, 10, 13, 15, 25], a)"""
        
    def test_big_bubble_sort(self):
        l = [x for x in range(20000, 0, -1)]
        bubble_sort.bubble_sort(l)
    
if __name__ == '__main__':
    unittest.main()
    
